### ISSUE TITLE
Biais on validation results due to fixed max_det value

### ISSUE THREAD

**Issue by aj0218700** [ultralytics/ultralytics#3043]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Validation

### Bug

In validation of training, the number of detections is limited to 15, which causes inconsistencies on results images if we have a large number of classes.
We should be able to customize the number of maximum detected classes, with an option to unlimited.
  
This is due to the '15' magic number here : 
https://github.com/ultralytics/ultralytics/blob/8940a27bdb26895f09a1554514a9a46312aa89c3/ultralytics/yolo/v8/detect/val.py#L15,  we have a magic value 15 here : output_to_target(preds, max_det=15).

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@aj0218700 great news 😃! Your original issue may now be resolved ✅ in PR #3051. 

To get this update:
- **[Git](https://github.com/ultralytics/ultralytics)** – Run `git pull` from within your `ultralytics/` directory or run `git clone https://github.com/ultralytics/ultralytics` again
- **[Pip](https://pypi.org/project/ultralytics/)** – Update with `pip install -U ultralytics`
- **[Notebooks](https://github.com/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb)** – Check out the updated notebooks  <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/ultralytics)** – Run `sudo docker pull ultralytics/ultralytics:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and letting us know. Please confirm if this update fixes the issue for you, and don't hesitate to report any other **issues** you find or **feature requests** you may have. Happy training with YOLOv8 🚀!

**Comment by Laughing-q** [comment]
@aj0218700 hi, thanks for the reporting! we set it 15 because plotting masks while validating instance segmentation model would cause extra time and memory, there are 300x640x640 masks need to be transferred to cpu for plotting which could stuck the process for a few seconds and also cost unnecessary memory up, and it would cost more time and memory when users train model with higher image size.
To keep consistency we set the same value for detection and pose tasks too, probably we should set 15 for segmentation task only.
